packages/default/scss/datetime/_theme.scss

```diff
@@ -61,6 +63,15 @@
                 $datetime-select-border,
                 $datetime-select-gradient
             );
+
+            .k-link:hover {
```

Comment:
There's no need to scope the `.k-link:hover` styles in `.k-select`, move them one level up.
As an example, you can check the [layout file](https://github.com/telerik/kendo-themes/blob/develop/packages/default/scss/datetime/_layout.scss#L151).

Reply:
I have moved it and also added the styles to the `.k-dateinput` scope...

However the hover styles aren't applied to the `.k-datepicker` and `.k-timepicker` because they don't contain `.k-link` elements inside the `.k-select`. This discrepancy in the HTML markup can be fixed by simply adding `k-link` class where it is missing to the `k-select`  wrapper class list.

Another way to solve this hover issue is with this temporary fix, to change the font color of the icon:
```
.k-link:hover > .k-icon,
.k-select:hover > .k-icon {
     color: primary
 }
```